lib/__tests__/cli.test.mjs

```diff
@@ -352,9 +362,10 @@ describe('CLI', () => {
 			fixturesPath('empty-block.css'),
 		]);
 
-		expect(process.exitCode).toBe(2);
+		expect(process.exitCode).toBeUndefined();
```

Comment:
question

Are you sure it's what was intended?
@hudochenkov do you confirm?
cf https://github.com/stylelint/stylelint/pull/5542/files#diff-0378929f7797bb789a616f16b2a0f5d0e53a28eba60469829002b0b30d60886cR324

Reply:
I guess exit code should be 0. Because warnings are ignored and there is no other violations.

Reply:
The actual exit code is 0 if we run the CLI in a terminal. I did some testing on my machine to verify that.

On the latest `v16` branch:

```console
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# git branch --show-current
v16
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# git rev-parse HEAD
9827e0d9cef369b8af097e97ccbdc32c8b454fd3
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# node bin/stylelint.mjs --quiet --config lib/__tests__/fixtures/default-severity-warning.json lib/__tests__/fixtures/empty-block.css
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# echo $?
0
```

On the `6604-code-stdout-error-stderr-v16` branch (the branch of this PR):

```console
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# git branch --show-current
6604-code-stdout-error-stderr-v16
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# git rev-parse HEAD
0f5f531ffe54bba6dfb44ec0e62ab99538b26b83
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# node bin/stylelint.mjs --quiet --config lib/__tests__/fixtures/default-severity-warning.json lib/__tests__/fixtures/empty-block.css
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# echo $?
0
```

Using `stylelint@latest`:

```console
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# npx stylelint@latest --quiet --config lib/__tests__/fixtures/default-severity-warning.json lib/__tests__/fixtures/empty-block.css
root@LAPTOP-7C2UU06T:~/my-fork/stylelint# echo $?
0
```

In this test, `process.exitCode` is expected to be `undefined` because that is the default value of `process.exitCode` in Node.js ([Reference](https://nodejs.org/api/process.html#processexitcode_1)) and we also explicitly reset `process.exitCode` to `undefined` in the `afterEach` hook.